docs/proposals/operator-managed-infra.md

```diff
@@ -0,0 +1,118 @@
+# Operator Managed Infra and multi namespace
+Design doc: Not a feature yet
+
+Instead of managing all the infrastructure through terraform, the
+operator itself should be responsible for managing the Workload Identity
+bindings between the K8s Service Account and the Cloud Service Account/IAM role.
+
+Doing this, will make it possible to use Substratus resources in multiple
+namespaces without having to use another tool to configure Cloud identity/role mappings.
+
+Out of scope:
+* Managing the bucket, image registry or other any other resources
+
+
+## Why?
+* Simplified install: Allow install to use `gcloud/aws` or `eksctl` to create initial K8s cluster + nodepools
+* Make install onto existing K8s cluster straight forward and remove
+  complexity of using Terraform
+* Improved UX, all an end-user needs to do is install our manifests and ensure GCP service account has enough permissions
+* Enterprises generally have very specific requirements about how they create their K8s Clusters and nodepools.
+* Allow Substratus to work in a multi namespace environment. Currently namespace is hardcoded in workload identity settings in the terraform
+
+## How?
+
+<img src="../diagrams/identity-global.excalidraw.png"></img>
+
+### Installation
+**Installation Flow** will be responsible for the following:
+* Create a K8s cluster with nodes/GPUs, this step is optional if end-user already has a K8s cluster (eksctl or gcloud)
+* Create Bucket, this step is optional if end-user already has a bucket
+* Create Image Registry, this step is optional if end-user already has an image registry
+* REQUIRED: Create Google Service Account or AWS IAM Role that has resource level permissions to:
+  * read and write a specific bucket
+  * pull and push to a specific Image Registry (GAR/ECR)
+  * (GCP) set IAM policy on the service Account itself to be able to manage workload identity bindings to KSAs. This can be done by assigning the service account to be a ServiceAccountAdmin to itself
+  * (AWS) UpdateAssumeRolePolicy on the AWS Role used by Substratus (required for multi namespace support)
+* REQUIRED: Apply Substratus manifests and configure them to use correct bucket and image registry
+
+### Controller
+Controller will be responsible for the following:
+* Creating the K8s ServiceAccount (already done today) and related binding by calling `enforceServiceAccount` whenever a Substratus resource gets reconciled
```

Comment:
To make the call provider agnostic, I was leaning toward `UpdateIamPrincipal` as the GRPC service we'd implement. The term principal seems to be common terminology across all providers. `Update` is in line with other verbs I've seen in GRPC-land.

```suggestion
* Creating the K8s ServiceAccount (already done today) and related binding by calling `UpdateIamPrincipal` whenever a Substratus resource gets reconciled
```

Variations:

* `UpdateIamPrincipalPolicy`
* `UpdateIdentityPolicy`

Reply:
I hadn't thought of this being a generic call but that makes total sense.

Reply:
I will rephrase this since we should re-use the existing enforceServiceAccount function but in that function i should call the generic `UpdateIamPrincipal` from `sci-${CLOUD}`